mentoring/components/questionnaire.py

```diff
@@ -0,0 +1,237 @@
+                # This list attribute is just a string of comma separated strings:
+                setattr(block, name, [unicode(val).strip() for val in value.split(',')])
+            elif isinstance(field, String):
+                setattr(block, name, value)
+            else:
+                setattr(block, name, field.from_json(value))
+
+        for xml_child in node:
+            if xml_child.tag is not etree.Comment:
+                block.runtime.add_node_as_child(block, xml_child, id_generator)
+
+        return block
+
+    @property
+    def studio_display_name(self):
+        return u"Question {}".format(self.step_number) if not self.lonely_step else u"Question"
+
+    def __getattribute__(self, name):
+        """ Provide a read-only display name without adding a display_name field to the class. """
+        if name == "display_name":
+            return self.studio_display_name
+        return super(QuestionnaireAbstractBlock, self).__getattribute__(name)
+
+    def student_view(self, context=None):
+        name = getattr(self, "unmixed_class", self.__class__).__name__
+
+        template_path = 'templates/html/{}.html'.format(name.lower())
+
+        context = context or {}
+        context['self'] = self
+        context['custom_choices'] = self.custom_choices
+
+        fragment = Fragment(loader.render_template(template_path, context))
+        fragment.add_css_url(self.runtime.local_resource_url(self, 'public/css/questionnaire.css'))
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/questionnaire.js'))
+        fragment.initialize_js(name)
+        return fragment
+
+    def mentoring_view(self, context=None):
+        return self.student_view(context)
+
+    def assessment_step_view(self, context=None):
+        """
+        assessment_step_view is the same as mentoring_view, except its DIV will have a different
+        class (.xblock-v1-assessment_step_view) that we use for assessments to hide all the
+        steps with CSS and to detect which children of mentoring are "Steps" and which are just
+        decorative elements/instructions.
+        """
+        return self.mentoring_view(context)
+
+    @property
+    def custom_choices(self):
+        custom_choices = []
+        for child_id in self.children:
+            if child_isinstance(self, child_id, ChoiceBlock):
+                custom_choices.append(self.runtime.get_block(child_id))
+        return custom_choices
+
+    @property
+    def all_choice_values(self):
+        return [c.value for c in self.custom_choices]
+
+    @property
+    def human_readable_choices(self):
+        return [{"display_name": c.content, "value": c.value} for c in self.custom_choices]
+
+    @staticmethod
+    def choice_values_provider(question):
+        """
+        Get a list a {"display_name": "Choice Description", "value": value}
+        objects for use with studio_view editor.
+        """
+        return question.human_readable_choices
```

Comment:
If there are currently no choices under the block, it renders as the old editor. This is problematic-- a confused user might fill in this field before adding choices, and cause a weird state.